packages/blade/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,185 @@
+import styled from 'styled-components';
+import type { ReactElement } from 'react';
+import { ProgressBarFilled } from './ProgressBarFilled';
+import { FormLabel } from '~components/Form';
+import { makeAccessible, makeSize } from '~utils';
+import { Text } from '~components/Typography/Text';
+import { useId } from '~src/hooks/useId';
+import { useTheme } from '~components/BladeProvider';
+import Box from '~components/Box';
+import type { ColorContrastTypes, Feedback } from '~tokens/theme/theme';
+
+type ProgressBarCommonProps = {
+  /**
+   * Sets aria-label to help users know what the progress bar is for. Default value is the same as the `label` passed.
+   */
+  accessibilityLabel?: string;
+  /**
+   * Sets the contrast for the progress bar
+   * @default 'low'
+   */
+  contrast?: ColorContrastTypes;
+  /**
+   * Sets the intent of the progress bar which changes the feedback color.
+   */
+  intent?: Feedback;
+  /**
+   * Sets the label to be rendered for the progress bar. This value will also be used as default for `accessibilityLabel`.
+   */
+  label?: string;
+  /**
+   * Sets the size of the progress bar.
+   * @default 'small'
+   */
+  size?: 'small' | 'medium';
+  /**
+   * Sets the progress value of the progress bar.
+   * @default 'small'
+   */
+  value?: number;
+};
+
+type ProgressBarVariant = 'progress' | 'meter';
+
+type ProgressBarProgressProps = ProgressBarCommonProps & {
+  /**
+   * Sets the variant to be rendered for thr progress bar.
+   * @default 'progress'
+   */
+  variant?: Extract<ProgressBarVariant, 'progress'>;
+  /**
+   * Sets whether the progress bar is in an indeterminate state.
+   * @default false
+   */
+  isIndeterminate?: boolean;
+  /**
+   * Sets whether or not to show the progress percentage for the progress bar. Percentage is hidden by default for the `meter` variant.
+   * @default false
+   */
+  hidePercentage?: boolean;
+};
+
+type ProgressBarMeterProps = ProgressBarCommonProps & {
+  /**
+   * Sets the variant to be rendered for thr progress bar.
+   * @default 'progress'
+   */
+  variant?: Extract<ProgressBarVariant, 'meter'>;
+  /**
+   * Sets whether the progress bar is in an indeterminate state.
+   * @default false
+   */
+  isIndeterminate?: undefined;
+  /**
+   * Sets whether or not to show the progress percentage for the progress bar. Percentage is hidden by default for the `meter` variant.
+   * @default false
+   */
+  hidePercentage?: undefined;
+};
+
+type ProgressBarProps = ProgressBarProgressProps | ProgressBarMeterProps;
+
+type ProgressBarUnfilledProps = {
+  backgroundColor: string;
+};
+
+const ProgressBarUnfilled = styled(Box)<ProgressBarUnfilledProps>(({ backgroundColor }) => ({
+  backgroundColor,
+}));
+
+const getProgress = (value: number): number => {
+  return Math.floor(Math.min(100, Math.max(0, value)));
```

Comment:
Optional: logic is bit complex to understand here, might be more readable to split this and replace with variables or add some comments

Reply:
Or also be better if we rename it to something like `clamp` or `clampValue` since it's clamping the value from 0 to 100

Reply:
Lodash has clamp and its documented well. Used that instead. Removed the `getProgress` function